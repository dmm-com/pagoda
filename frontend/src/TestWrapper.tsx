import { createTheme, ThemeProvider } from "@mui/material/styles";
import { SnackbarProvider } from "notistack";
import React, { FC } from "react";
import { MemoryRouter } from "react-router";

export const TestWrapper: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        <MemoryRouter>{children}</MemoryRouter>
      </SnackbarProvider>
    </ThemeProvider>
  );
};

export const TestWrapperWithoutRoutes: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        {children}
      </SnackbarProvider>
    </ThemeProvider>
  );
};
