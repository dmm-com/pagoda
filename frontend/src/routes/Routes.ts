export const basePath = "/ui/";

export const loginPath = () => "/auth/login/";
export const topPath = () => basePath;
export const advancedSearchPath = () => basePath + "advanced_search";
export const advancedSearchResultPath = () =>
  basePath + "advanced_search_result";
export const jobsPath = (targetId?: number) =>
  basePath + "jobs" + (targetId ? `?target_id=${targetId}` : "");
export const aclPath = (objectId: number | string) =>
  basePath + `acl/${objectId}`;
export const aclHistoryPath = (objectId: number | string) =>
  basePath + `acl/${objectId}/history`;
export const searchPath = () => basePath + "search";

// entries
export const newEntryPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/entries/new`;
export const copyEntryPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/copy`;
export const entryDetailsPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/details`;
export const entryEditPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/edit`;
export const entityEntriesPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/entries`;
export const restoreEntryPath = (entityId: number | string, keyword?: string) =>
  basePath +
  `entities/${entityId}/restore${keyword != null ? "?query=" + keyword : ""}`;
export const showEntryHistoryPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/history`;

// entities
export const entityHistoryPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/history`;
export const listAliasPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/alias`;
export const newEntityPath = () => basePath + "entities/new";
export const editEntityPath = (entityId: number | string) =>
  basePath + `entities/${entityId}`;
export const entitiesPath = () => basePath + "entities";

// triggers
export const newTriggerPath = () => basePath + "triggers/new";
export const editTriggerPath = (triggerId: number | string) =>
  basePath + `triggers/${triggerId}`;
export const triggersPath = () => basePath + "triggers";

// groups
export const newGroupPath = () => basePath + "groups/new";
export const groupPath = (groupId: number | string) =>
  basePath + `groups/${groupId}`;
export const groupsPath = () => basePath + "groups";

// users
export const newUserPath = () => basePath + "users/new";
export const userPath = (userId: number | string) =>
  basePath + `users/${userId}`;
export const usersPath = () => basePath + "users";

// roles
export const newRolePath = () => basePath + "roles/new";
export const rolePath = (roleId: number | string) =>
  basePath + `roles/${roleId}`;
export const rolesPath = () => basePath + "roles";


// category
export const newCategoryPath = () => basePath + "categories/new";
export const listCategoryPath = () => basePath + "categories/list";
export const editCategoryPath = (categoryId: number | string) => basePath + "categories/edit";