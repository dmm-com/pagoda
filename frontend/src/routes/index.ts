export * from "./AppRouter";
export * from "./Routes";
