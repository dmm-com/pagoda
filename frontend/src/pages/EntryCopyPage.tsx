import { Box, Container } from "@mui/material";
import { useSnackbar } from "notistack";
import React, { FC, useState } from "react";
import { useNavigate } from "react-router";

import { useAsyncWithThrow } from "../hooks/useAsyncWithThrow";
import { useTypedParams } from "../hooks/useTypedParams";

import { Loading } from "components/common/Loading";
import { PageHeader } from "components/common/PageHeader";
import { SubmitButton } from "components/common/SubmitButton";
import {
  CopyForm as DefaultCopyForm,
  CopyFormProps,
} from "components/entry/CopyForm";
import { EntryBreadcrumbs } from "components/entry/EntryBreadcrumbs";
import { usePrompt } from "hooks/usePrompt";
import { aironeApiClient } from "repository/AironeApiClient";
import { entityEntriesPath, entryDetailsPath } from "routes/Routes";

interface Props {
  CopyForm?: FC<CopyFormProps>;
}

export const EntryCopyPage: FC<Props> = ({ CopyForm = DefaultCopyForm }) => {
  const navigate = useNavigate();
  const { enqueueSnackbar } = useSnackbar();
  const { entityId, entryId } = useTypedParams<{
    entityId: number;
    entryId: number;
  }>();

  // newline delimited string value, not string[]
  const [entries, _setEntries] = useState<string>("");
  const [submitting, setSubmitting] = useState<boolean>(false);
  const [submitted, setSubmitted] = useState<boolean>(false);
  const [edited, setEdited] = useState<boolean>(false);

  usePrompt(
    edited && !submitted,
    "編集した内容は失われてしまいますが、このページを離れてもよろしいですか？",
  );

  const entry = useAsyncWithThrow(async () => {
    return await aironeApiClient.getEntry(entryId);
  }, [entryId]);

  if (entry.loading) {
    return <Loading />;
  }

  const setEntries = (entries: string) => {
    setEdited(true);
    _setEntries(entries);
  };

  const handleCopy = async () => {
    setSubmitting(true);
    try {
      await aironeApiClient.copyEntry(
        entryId,
        entries
          .split("\n")
          .map((e) => e.trim())
          .filter((e) => e.length > 0),
      );

      setSubmitted(true);
      enqueueSnackbar("アイテムコピーのジョブ登録が成功しました", {
        variant: "success",
      });
      setTimeout(() => {
        navigate(entityEntriesPath(entityId), { replace: true });
      }, 0.1);
    } catch {
      enqueueSnackbar("アイテムコピーのジョブ登録が失敗しました", {
        variant: "error",
      });
    }
  };

  const handleCancel = () => {
    navigate(
      entryDetailsPath(entry.value?.schema?.id ?? 0, entry.value?.id ?? 0),
      { replace: true },
    );
  };

  return (
    <Box>
      <EntryBreadcrumbs entry={entry.value} title="コピー" />

      <PageHeader
        title={entry.value?.name ?? ""}
        description="アイテムのコピーを作成"
      >
        <SubmitButton
          name="コピーを作成"
          disabled={!entries || submitting || submitted}
          isSubmitting={submitting}
          handleSubmit={handleCopy}
          handleCancel={handleCancel}
        />
      </PageHeader>

      <Container>
        {entry.value && (
          <CopyForm
            entries={entries}
            setEntries={setEntries}
            templateEntry={entry.value}
          />
        )}
      </Container>
    </Box>
  );
};
