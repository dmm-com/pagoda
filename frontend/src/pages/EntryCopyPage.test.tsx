/**
 * @jest-environment jsdom
 */

import { render, screen, act, waitFor } from "@testing-library/react";
import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";
import React from "react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";

import { EntryCopyPage } from "./EntryCopyPage";

import { TestWrapperWithoutRoutes } from "TestWrapper";
import { copyEntryPath } from "routes/Routes";

const server = setupServer(
  // getEntry
  http.get("http://localhost/entry/api/v2/1/", () => {
    return HttpResponse.json({
      id: 1,
      name: "test entry",
      is_active: true,
      schema: {
        id: 2,
        name: "test entity",
      },
      attrs: [],
    });
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("should match snapshot", async () => {
  const router = createMemoryRouter(
    [
      {
        path: copyEntryPath(":entityId", ":entryId"),
        element: <EntryCopyPage />,
      },
    ],
    {
      initialEntries: [copyEntryPath(2, 1)],
    }
  );
  const result = await act(async () => {
    return render(<RouterProvider router={router} />, {
      wrapper: TestWrapperWithoutRoutes,
    });
  });
  await waitFor(() => {
    expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
  });

  expect(result).toMatchSnapshot();
});
