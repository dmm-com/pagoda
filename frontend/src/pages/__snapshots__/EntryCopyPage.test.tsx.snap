// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1baykhj"
        >
          <div
            class="MuiBox-root css-t0pdf1"
          >
            <div
              class="MuiBox-root css-jqffe8"
            >
              <nav
                aria-label="breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      href="/ui/"
                    >
                      Top
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      href="/ui/entities"
                    >
                      モデル一覧
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <div
                      class="MuiBox-root css-gvoll6"
                    >
                      <a
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                        href="/ui/entities/2/entries"
                      >
                        test entity
                      </a>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LockIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <div
                      class="MuiBox-root css-gvoll6"
                    >
                      <a
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                        href="/ui/entities/2/entries/1/details"
                      >
                        test entry
                      </a>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LockIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
                    >
                      コピー
                    </p>
                  </li>
                </ol>
              </nav>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1wo2fgg"
        >
          <div
            class="MuiBox-root css-t1gim7"
          >
            <div
              class="MuiBox-root css-9rtbay"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-178toxd-MuiTypography-root"
                id="title"
              >
                test entry
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                id="description"
              >
                アイテムのコピーを作成
              </h6>
              <div
                class="MuiBox-root css-1b3gh38"
              >
                <div
                  class="MuiBox-root css-1wcjc1k"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-jjpace-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    コピーを作成
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium css-ljnmqm-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    キャンセル
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
            />
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            入力した各行ごとに test entry と同じ属性を持つ別のアイテムを作成
          </p>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-347vvv-MuiInputBase-input-MuiOutlinedInput-input"
                id="copy-name"
                placeholder="コピーするアイテム名"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-347vvv-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-1xioh3b"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-41zwy8-MuiTypography-root"
            >
              SAMPLE
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              (Vm0001、vm0002、…vm006の6アイテムを作成する場合)
            </p>
            <div
              class="MuiFormControl-root MuiTextField-root css-ksw8wi-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for=":r1:"
                id=":r1:-label"
              >
                コピーするアイテム名
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id=":r1:"
                  style="height: 0px; overflow: hidden;"
                >
                  vm0001
vm0002
vm0003
vm0004
vm0005
vm0006
                </textarea>
                <textarea
                  aria-hidden="true"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  readonly=""
                  style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                  tabindex="-1"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      コピーするアイテム名
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-1baykhj"
      >
        <div
          class="MuiBox-root css-t0pdf1"
        >
          <div
            class="MuiBox-root css-jqffe8"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    href="/ui/"
                  >
                    Top
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    href="/ui/entities"
                  >
                    モデル一覧
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <div
                    class="MuiBox-root css-gvoll6"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      href="/ui/entities/2/entries"
                    >
                      test entity
                    </a>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="LockIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <div
                    class="MuiBox-root css-gvoll6"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      href="/ui/entities/2/entries/1/details"
                    >
                      test entry
                    </a>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="LockIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
                  >
                    コピー
                  </p>
                </li>
              </ol>
            </nav>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1wo2fgg"
      >
        <div
          class="MuiBox-root css-t1gim7"
        >
          <div
            class="MuiBox-root css-9rtbay"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-178toxd-MuiTypography-root"
              id="title"
            >
              test entry
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              id="description"
            >
              アイテムのコピーを作成
            </h6>
            <div
              class="MuiBox-root css-1b3gh38"
            >
              <div
                class="MuiBox-root css-1wcjc1k"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-jjpace-MuiButtonBase-root-MuiButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  コピーを作成
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium css-ljnmqm-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  キャンセル
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
          />
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          入力した各行ごとに test entry と同じ属性を持つ別のアイテムを作成
        </p>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-347vvv-MuiInputBase-input-MuiOutlinedInput-input"
              id="copy-name"
              placeholder="コピーするアイテム名"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-347vvv-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-1xioh3b"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-41zwy8-MuiTypography-root"
          >
            SAMPLE
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
          >
            (Vm0001、vm0002、…vm006の6アイテムを作成する場合)
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root css-ksw8wi-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r1:"
              id=":r1:-label"
            >
              コピーするアイテム名
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r1:"
                style="height: 0px; overflow: hidden;"
              >
                vm0001
vm0002
vm0003
vm0004
vm0005
vm0006
              </textarea>
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    コピーするアイテム名
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
